github_discovery.py

```diff
@@ -249,8 +290,22 @@ def process_repo(**component):
       if file.name.startswith('values-'):
         env = re.match('values-([a-z0-9-]+)\\.y[a]?ml', file.name)[1]
         helm_environments.append(env)
+
+        # HEAT-223 Start : Read and collate data for allowlist from all environment specific values.yaml files.
+        ip_allow_list[file] = fetch_values_for_allowlist_key(get_file_yaml(repo, f"{monorepo_dir_suffix}helm_deploy/{file.name}"), allow_list_key) 
```

Comment:
this is duplicating code on line 299. Suggest you move this block after 299. and do 
```suggestion
        ip_allow_list[file] = fetch_values_for_allowlist_key(helm_default_values, allow_list_key) 
```